access-grant/src/main/java/com/inrupt/client/accessgrant/AccessRequest.java

```diff
@@ -123,7 +123,8 @@ static AccessRequest parse(final String serialization) throws IOException {
                 final Set<String> modes = asSet(consent.get("mode")).orElseGet(Collections::emptySet);
                 final Set<URI> resources = asSet(consent.get("forPersonalData")).orElseGet(Collections::emptySet)
                     .stream().map(URI::create).collect(Collectors.toSet());
-                final Set<String> purposes = asSet(consent.get("forPurpose")).orElseGet(Collections::emptySet);
+                final Set<URI> purposes = asSet(consent.get("forPurpose")).orElseGet(Collections::emptySet)
+                    .stream().map(URI::create).collect(Collectors.toSet());
```

Comment:
`.map(URI::create)` might fail, no?

Reply:
Addressed in https://github.com/inrupt/solid-client-java/pull/491/commits/3b79f521c2335cd4078ed54e706b5e9b7a4a81fe